### PR TITLE
[improvement](jdbc) refator jdbc of copy result set by batch

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

`have test jdbc external table with read,  10%+ performance improvement after optimization`

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.24 seconds
 load time: 497 seconds
 storage size: 17123263303 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230201191328_clickbench_pr_88920.html

File: fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java [hunk 1]
```diff
@@ -325,277 +325,233 @@ private void init(String driverUrl, String sql, int batchSize, String driverClas
     public void copyBatchBooleanResult(Object columnObj, boolean isNullable, int numRows, long nullMapAddr,
             long columnAddr) {
         Boolean[] column = (Boolean[]) columnObj;
-        byte[] columnData = new byte[numRows];
         if (isNullable) {
-            byte[] nullMap = new byte[numRows];
             for (int i = 0; i < numRows; i++) {
                 if (column[i] == null) {
-                    nullMap[i] = 1;
+                    UdfUtils.UNSAFE.putByte(nullMapAddr + i, (byte) 1);
```
Comment (HappenLee) [inline fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java:319]: should always putByte. if column[i] != null, put 0

Comment (zhangstar333) [reply fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java:312]: add memset(null_map.data(), 0, num_rows) in be

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**